javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java

```diff
@@ -105,8 +105,8 @@ public ClientModelProperty map(Property property) {
                 && sequence.getElementType().getSerialization().getXml() != null
                 && sequence.getElementType().getSerialization().getXml().getName() != null) {
                 builder.xmlListElementName(sequence.getElementType().getSerialization().getXml().getName());
-            } else if(sequence.getSerialization() != null && sequence.getSerialization().getXml() != null){
-                builder.xmlListElementName(sequence.getSerialization().getXml().getName());
+            } else {
+                builder.xmlListElementName(sequence.getElementType().getLanguage().getDefault().getName());
```

Comment:
@srnagar if serialization.xml.name isn't available - it doesn't seem right to use the array schema's xml name for the item name. For now I'm using the name of the item as a fallback and seems to generate this correctly so far.

Reply:
Sounds good.